discord/ext/commands/core.py

```diff
@@ -908,20 +919,22 @@ def signature(self):
                 # do [name] since [name=None] or [name=] are not exactly useful for the user.
                 should_print = param.default if isinstance(param.default, str) else param.default is not None
                 if should_print:
-                    result.append('[%s=%s]' % (name, param.default) if not greedy else
-                                  '[%s=%s]...' % (name, param.default))
+                    result_params.append('[%s=%s]' % (name, param.default) if not greedy else
+                                         '[%s=%s]...' % (name, param.default))
                     continue
                 else:
-                    result.append('[%s]' % name)
+                    result_params.append('[%s]' % name)
 
             elif param.kind == param.VAR_POSITIONAL:
-                result.append('[%s...]' % name)
+                result_params.append('[%s...]' % name)
             elif greedy:
-                result.append('[%s]...' % name)
+                result_params.append('[%s]...' % name)
             elif self._is_typing_optional(param.annotation):
                 result.append('[%s]' % name)
```

Comment:
Shouldn't this also be `result_params.append`?

Likewise, what's the point of separating `result` and `result_params` if only one ends up being used?